client/src/components/papi.js

```diff
@@ -11,12 +11,23 @@ export default function Papi({ selectedPatientID }) {
         calculatedValue: ""
     });
     const [placeholderText, setPlaceholderText] = useState("");
+    const [error, setError] = useState(null);
 
     useEffect(() => {
+        if (ra === 0) {
+            setError('RA cannot be zero');
+            setForm({ valueType: "Pulmonary Artery Pulsatility Index", calculatedValue: "" });
+            return;
+        }
+        
         setPlaceholderText((pasp === "" && padp === "" && ra === "") ? "Enter calculation inputs" : "Missing inputs");
         setForm({ valueType: "Pulmonary Artery Pulsatility Index", calculatedValue: (pasp !== "" && padp !== "" && ra !== "") ? ((+pasp - +padp) / +ra).toFixed(3) : "" });
     }, [pasp, padp, ra]);
 
+    if (error) {
+        return <div>Error: {error}</div>;
```

Comment:
I think it'd be more helpful if a message was displayed or if the user was notified rather than removing the calculator and showing the error message. With the current implementation, there is no way to re-enable the calculation without restarting the application. Once an error results, the error message is stuck there for that calculation.